components/checkout/pay/SavedCardDisplay.vue

```diff
@@ -0,0 +1,226 @@
+  emit('update:initializing', valid);
+}
+
+const { client, card } = toRefs(props);
+const css = useCssModule();
+
+function setupVaults() {
+  let _vaults: VaultManager | null = null;
+
+  const getVaults = () => {
+    assert(_vaults);
+    return _vaults;
+  };
+
+  const create = async (client: Client) => {
+    _vaults = await vaultManager.create({
+      client,
+    });
+  };
+
+  const teardown = () => {
+    _vaults?.teardown();
+  };
+
+  return {
+    create,
+    get: getVaults,
+    focus,
+    teardown,
+  };
+}
+
+const vaults = setupVaults();
+
+const { t } = useI18n();
+
+const submitPayload: Ref<{ nonce: string; bin: string }> = ref({
+  nonce: '',
+  bin: '',
+});
+const payloadError: Ref<string> = ref('');
+
+onMounted(async () => {
+  try {
+    updateInitializing(true);
+    await vaults.create(get(client));
+    await checkPaymentMethods();
+  }
+  catch (error_: any) {
+    updateError({
+      title: t('subscription.error.init_error'),
+      message: error_.message,
+    });
+  }
+  finally {
+    updateInitializing(false);
+  }
+});
+
+watchImmediate(payloadError, (payloadError) => {
+  updateFormValid(!payloadError);
+});
+
+onUnmounted(() => {
+  vaults.teardown();
+});
+
+async function checkPaymentMethods() {
+  const methods = await vaults.get().fetchPaymentMethods();
+
+  const method = methods.find(({ type, details }) => type === 'CreditCard' && details && 'lastFour' in details && details?.lastFour === get(card).last4);
+
+  if (!method?.details
+    || !('bin' in method.details)
+    || !('expirationMonth' in method.details)
+    || !('expirationYear' in method.details)
+  ) {
+    set(payloadError, t('home.plans.tiers.step_3.saved_card.cannot_use_card'));
+  }
+  else {
+    const current = new Date();
+    const month = current.getMonth() + 1;
+    const year = current.getFullYear();
+    const expirationMonth = Number(method.details.expirationMonth);
+    const expirationYear = Number(method.details.expirationYear);
+
+    if (expirationYear > year || (expirationYear === year && expirationMonth >= month)) {
+      const nonce = method.nonce;
+      const bin = method.details.bin;
+
+      set(submitPayload, { nonce, bin });
+    }
+    else {
+      set(payloadError, t('home.plans.tiers.step_3.saved_card.card_expired'));
+    }
+  }
+}
+
+function submit() {
+  return get(submitPayload);
+}
+
+defineExpose({ submit });
+
+const { deleteCard } = usePaymentCardsStore();
+
+const deleteLoading = ref(false);
+
+const showDeleteConfirmation = ref(false);
+
+function deleteCardClick() {
+  set(showDeleteConfirmation, true);
+}
+
+async function handleDeleteCard() {
+  set(showDeleteConfirmation, false);
+  set(deleteLoading, true);
+  await deleteCard(get(card).token);
+  set(deleteLoading, false);
+}
+</script>
+
+<template>
+  <div :class="css.card">
+    <div class="rounded-md bg-rui-grey-50 h-10 w-14 flex items-center justify-center">
+      <img
+        :src="card.imageUrl"
```

Comment:
we need to append `assets.braintreegateway.com/`  to `img-src` in the CSP for card payment I don't thing we already have it.

https://github.com/rotki/rotki.com/blob/b502ea1d5873a0c16615ea18d5a9102c70c9b5c4/server/middleware/csp.ts#L73

```typescript
    finalPolicy[ContentPolicy.IMG_SRC] = [
      ...finalPolicy[ContentPolicy.IMG_SRC],
      'assets.braintreegateway.com',
      ]

```